src/Components/Host.php

```diff
@@ -266,7 +266,15 @@ public function getLabels()
     public function getLabel($offset, $default = null)
     {
         if (isset($this->data[$offset])) {
-            return $this->isIdn ? rawurldecode($this->data[$offset]) : idn_to_ascii($this->data[$offset]);
+            if ($this->isIdn) {
+                return rawurldecode($this->data[$offset]);
+            }
+
+            if (PHP_VERSION_ID >= 70200) {
```

Comment:
Adding suppression operator seems  not really good to me this library requires PHP5.5.9 and Intl so you can directly use :

`idn_to_utf8($part, 0, INTL_IDNA_VARIANT_UTS46)`
and 
`idn_to_ascii($part, 0, INTL_IDNA_VARIANT_UTS46)`

since Pdp dependencies will also be using this variant anyway.

Reply:
I've tried to do so and two tests are failing then:

```
There were 2 errors:

1) League\Uri\Test\Components\HostTest::testValidHost with data set "invalid punycode" ('xn--fsqu00a.xn--g6w131251d', false, false, false, 'xn--fsqu00a.xn--g6w131251d')
InvalidArgumentException: Invalid Hostname, empty labels are not allowed

/Users/pamil/Projects/League/uri/src/Components/HostnameTrait.php:116
/Users/pamil/Projects/League/uri/src/Components/HostnameTrait.php:71
/Users/pamil/Projects/League/uri/src/Components/Host.php:149
/Users/pamil/Projects/League/uri/src/Components/Host.php:122
/Users/pamil/Projects/League/uri/test/Components/HostTest.php:49

2) League\Uri\Test\Components\HostTest::testValidUnicodeHost with data set #21 ('例子.xn--1', 'xn--fsqu00a.xn--1')
InvalidArgumentException: Invalid Hostname, empty labels are not allowed

/Users/pamil/Projects/League/uri/src/Components/HostnameTrait.php:116
/Users/pamil/Projects/League/uri/src/Components/HostnameTrait.php:71
/Users/pamil/Projects/League/uri/src/Components/Host.php:149
/Users/pamil/Projects/League/uri/src/Components/Host.php:122
/Users/pamil/Projects/League/uri/test/Components/HostTest.php:161

FAILURES!
Tests: 912, Assertions: 1235, Errors: 2.
```

Reply:
those tests are failing because they are using the old algorithm . so the tests need to be updated the deprecated algorithm is not a PHP decision but an ICU one see https://externals.io/message/97164

Reply:
Not sure if I understand correctly, should these tests cases be removed then?

Reply:
Using `INTL_IDNA_VARIANT_UTS46` will make this library to switch its behaviour in a patch release, so I'd rather just silence errors on PHP 7.2.

Currently it crashes on PHP 7.2 and with this fix it will stop crashing until PHP 7.4 changes the default variant value, so I think it's good enough and the applications relying on the current `3.x` version will continue working the same.

Reply:
you are right :+1:

Reply:
All right, falling back to the previous solution then :)